src/Tests/Microsoft.DotNet.GenAPI.Tests/SyntaxRewriterTests.cs

```diff
@@ -0,0 +1,385 @@
+﻿// Copyright (c) .NET Foundation and contributors. All rights reserved.
+// Licensed under the MIT license. See LICENSE file in the project root for full license information.
+
+#nullable enable
+
+using Xunit;
+using System.IO;
+using System.Text;
+using Microsoft.CodeAnalysis.CSharp;
+using Microsoft.DotNet.GenAPI.SyntaxRewriter;
+using Microsoft.CodeAnalysis;
+
+namespace Microsoft.DotNet.GenAPI.Tests
+{
+    public class SyntaxRewriterTests
+    {
+        protected void Compare(CSharpSyntaxRewriter rewriter, string original, string expected)
+        {
+            StringWriter _stringWriter = new();
+            SyntaxNode root = CSharpSyntaxTree.ParseText(original).GetRoot();
+            rewriter.Visit(root)
+                .WriteTo(_stringWriter);
+
+            StringBuilder stringBuilder = _stringWriter.GetStringBuilder();
+            var resulted = stringBuilder.ToString();
+
+            Assert.True(resulted.Equals(expected),
+                $"Expected:\n{expected}\nResulted:\n{resulted}");
+        }
+
+        protected void CompareSyntaxTree(CSharpSyntaxRewriter rewriter, string original, string expected)
+        {
+            StringWriter _stringWriter = new();
+            SyntaxNode root = CSharpSyntaxTree.ParseText(original).GetRoot();
+            rewriter.Visit(root)
+                .WriteTo(_stringWriter);
+
+            StringBuilder stringBuilder = _stringWriter.GetStringBuilder();
+            var resulted = stringBuilder.ToString();
+
+            SyntaxTree resultedSyntaxTree = CSharpSyntaxTree.ParseText(resulted);
+            SyntaxTree expectedSyntaxTree = CSharpSyntaxTree.ParseText(expected);
+
+            // compare SyntaxTree and not string representation
+            Assert.True(resultedSyntaxTree.IsEquivalentTo(expectedSyntaxTree),
+                $"Expected:\n{expected}\nResulted:\n{resulted}");
+        }
+    }
+
+    public class OneLineStatementSyntaxRewriterTests : SyntaxRewriterTests
+    {
+        [Fact]
+        public void TestEmptyMethodBody()
+        {
+            Compare(new OneLineStatementSyntaxRewriter(),
+                original: """
+                namespace A
+                {
+                    class B
+                    {
+                        void Execute() {}
```

Comment:
Is it possible to make the method body contain a whitespace character? Asking, as the old GenAPI produced this output: https://github.com/dotnet/runtime/blob/c50c745fa88f0850c8e8ebed37749118929d7e87/src/libraries/System.Text.RegularExpressions/ref/System.Text.RegularExpressions.cs#L11

Reply:
I think we may control the behavior with Formatter options https://github.com/dotnet/sdk/blob/1d4cb25ff11a718bece0bdd540d6ba1a6f86a719/src/GenAPI/Microsoft.DotNet.GenAPI/CSharpFileBuilder.cs#L161-L174

or here `SingleLineStatementCSharpSyntaxRewriter` as a post processor. Will follow up on this.